api/src/main/java/org/pmiops/workbench/cohortbuilder/FieldSetQueryBuilder.java

```diff
@@ -42,14 +52,189 @@
 
   private final ParticipantCounter participantCounter;
 
+
+  /**
+   * A container for state needed to turn a query into SQL. We pass this around to avoid having
+   * to pass around a whole bunch of arguments.
+   */
+  private static final class QueryState {
+    // SQL for the select clause
+    private StringBuilder selectSql = new StringBuilder();
+    // SQL for the from clause
+    private StringBuilder fromSql = new StringBuilder();
+    // SQL for the where clause
+    private StringBuilder whereSql = new StringBuilder();
+    // SQL for order by, limit, offset
+    private StringBuilder endSql = new StringBuilder();
+    // A map of parameter values for bound parameters referenced in the query
+    private Map<String, QueryParameterValue> paramMap = new HashMap<>();
+    // A map from table name to (column name -> ColumnConfig) for tables referenced in the query.
+    private Map<String, Map<String, ColumnConfig>> columnConfigTable = new HashMap<>();
+    // A map from table aliases to SQL table names.
+    private Map<String, String> aliasToTableName = new HashMap<>();
+    // The columns selected in the select clause.
+    private ImmutableList.Builder<ColumnInfo> selectColumns = ImmutableList.builder();
+    // The query being turned into SQL.
+    private TableQuery tableQuery;
+    // The CDR schema configuration.
+    private CdrBigQuerySchemaConfig schemaConfig;
+    // A map of (column name -> ColumnConfig) for the columns on the main table (pulled from
+    // columnConfigTable).
+    private Map<String, ColumnConfig> mainTableColumns;
+  }
+
+  private static final class TableNameAndAlias {
+    private final String tableName;
+    private final String alias;
+
+    TableNameAndAlias(String tableName, String alias) {
+      this.tableName = tableName;
+      this.alias = alias;
+    }
+  }
+
   @Autowired
   public FieldSetQueryBuilder(ParticipantCounter participantCounter) {
     this.participantCounter = participantCounter;
```

Comment:
So ParticipantCounter was created to wrap any of the CB search criteria queries that were ORed or ANDed together and get either a count or ids from the person table. But now it seems that we are using this class to query any of the omop tables. So i think we should either split out materialized endpoint queries into it's own class and share code between the PariticpantCounter and this class or rename it to something more descriptive.

Reply:
I pulled the common query building code out into a new CohortQueryBuilder class, which is used from both ParticipantCounter and FieldSetQueryBuilder.